src/components/ha-data-table.ts

```diff
@@ -387,14 +396,21 @@ export class HaDataTable extends BaseElement {
 
   private _handleRowCheckboxChange(ev: Event) {
     const checkbox = ev.target as HaCheckbox;
-    const rowId = checkbox.parentElement!.parentElement!.getAttribute(
-      "data-row-id"
-    );
+    const rowId = checkbox.closest("tr")!.getAttribute("data-row-id");
 
     this._setRowChecked(rowId!, checkbox.checked);
     this.mdcFoundation.handleRowCheckboxChange(ev);
   }
 
+  private _handleRowClick(ev: Event) {
```

Comment:
Do we still need this row click if we are not using it ?

Reply:
It is used in the devices table.